wkcuber/convert_nifti.py

```diff
@@ -0,0 +1,179 @@
+import time
+import logging
+import wkw
+import numpy as np
+from argparse import ArgumentParser
+from pathlib import Path
+import nibabel as nib
+
+from .utils import (
+    add_verbose_flag,
+    open_wkw,
+    WkwDatasetInfo,
+    ensure_wkw,
+    setup_logging,
+)
+
+from .metadata import write_webknossos_metadata
+
+
+def create_parser():
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "source_path",
+        help="Path to NIFTY file or to a directory if multiple NIFTI files should be converted. "
+        "In the latter case, also see --color_file and --segmentation_file.",
+    )
+
+    parser.add_argument(
+        "target_path", help="Output directory for the generated WKW dataset."
+    )
+
+    parser.add_argument(
+        "--layer_name",
+        "-l",
+        help="Name of the cubed layer (color or segmentation).",
+        default="color",
+    )
+
+    parser.add_argument(
+        "--dtype",
+        "-d",
+        help="Target datatype (e.g. uint8, uint16, uint32).",
+        default="uint8",
+    )
+
+    parser.add_argument(
+        "--color_file",
+        help="When converting folder, name of file to become color layer",
+        default=None,
+    )
+
+    parser.add_argument(
+        "--segmentation_file",
+        help="When converting folder, name of file to become segmentation layer",
+        default=None,
+    )
+
+    parser.add_argument(
+        "--scale", help="The scale of the NIFTI input in the form X,Y,Z.", default=None,
```

Comment:
I am pretty sure `scale` is read as a string from the command-line, e.g. `"10, 10, 30"`. Instead it should be a list, e.g. `[10, 10, 30]`.

Reply:
Please add some type assertion to `write_webknossos_metadata` to prevent this in the future. 

See `(from .metadata import write_webknossos_metadata)`

Reply:
Not sure, if I asked before: Isn't the scale stored in the nifti header?

Reply:
The `main` and the `metadata` entry point also use a scale parameter which needs to be passed in the form of `10,10,10` (without any brackets). Therefore, convert_nifti should do the same.
However, you can import `add_scale_flag` from `utils.py` for that. Converting the scale from string to a list is done via `scale = tuple(float(x) for x in args.scale.split(","))` (see main method in `__main__`). It would be cool if you could make the scale-parsing part of the argparse definition. Something like this would do the trick:

```
def parse_scale(scale):
  try:
    scale = tuple(float(x) for x in args.scale.split(","))
    return scale
  except Exception:
        raise argparse.ArgumentTypeError("The scale could not be parsed")

def add_scale_flag(parser, required=True):
    parser.add_argument(
        "--scale",
        "-s",
        help="Scale of the dataset (e.g. 11.2,11.2,25). This is the size of one voxel in nm.",
        required=required,
        type=parse_scale,
    )
```

Then, `args.scale` doesn't need to be parsed for each usage of `add_scale_flag`.

Reply:
@leowe BTW, I already fixed the `datasource-properties.json` files for stroke-vision

Reply:
Okay, so there is a parameter in the nifty header that can store the scale of the dataset (`pixdim`). Should I make the scale flag optional and try to infer the scale from the nifty header and when the parameter is given simply overwrite the inferred scale?

Reply:
Sounds reasonable!

Reply:
I now implemented it as discussed.